hudi-utilities/src/main/java/org/apache/hudi/utilities/sources/ParquetDFSSource.java

```diff
@@ -52,6 +52,6 @@ public Pair<Option<Dataset<Row>>, String> fetchNextBatch(Option<String> lastCkpt
   }
 
   private Dataset<Row> fromFiles(String pathStr) {
-    return sparkSession.read().parquet(pathStr.split(","));
+    return sparkSession.read().option("mergeSchema", "true").parquet(pathStr.split(","));
```

Comment:
Can we add a config property for this and enable based on that. 
you can introduce a new Config class named ParquetDFSSourceConfig and add a config property for MergeSchema. 
set default to true. 

you can take a look at https://github.com/apache/hudi/blob/master/hudi-utilities/src/main/java/org/apache/hudi/utilities/config/DFSPathSelectorConfig.java for reference.